Redesign default.haml

svbergerem opened the pull request:
![default.haml](https://cloud.githubusercontent.com/assets/3798614/8956564/78abc4ee-35f7-11e5-8d42-2b15e8a15291.png)

Please comment if you have any idea what we should put on that page.

jhass commented on app/views/layouts/application.html.haml:
```diff
@@ -32,10 +32,8 @@
 
     = jquery_include_tag
-
-    - unless @landing_page
```
Still any other uses of that variable left? Else I'd drop it from the controller too, including the application JS will likely have some potential to break custom landing pages anyway.

svbergerem replied:
```bash
$ grep -Rin "landing" app config lib spec features
app/assets/stylesheets/mobile/mobile.scss:786:.landing  {
app/views/sessions/new.mobile.haml:6:  .landing
features/mobile/home.feature:2:Feature: Visit the landing page of the pod
features/mobile/home.feature:5:  I want to see the landing page
```

jhass replied:
So we never set it anymore? >.<

jhass replied:
yay, https://github.com/diaspora/diaspora/commit/b8f7d025d50744c9f4b65cacbf675927dd424938#diff-14dfef9f01460f69f175a8e6bd7d143dR21

Flaburgan commented:
The current default page is addressed to the podmin, telling him to configure his pod and contribute. It's nice to have a default page oriented to users, but that means the install guide has to provide the information we previously put there. Are we sure the wiki is enough to do that? Some pods are installed without following the install guide (with apt-get, with bitnami etc). Do we really want the content for the podmin to completely disappear from this page?

jhass commented:
Just throwing out an idea here, not sure how I feel about myself yet.

We could do two default pages, a podmin oriented one if no (or only one or two?) accounts are registered and switch to a user oriented default once there's an account.

pablocubico commented:
Just for you guys to know, this is the current design from my upcoming pod.

![screen shot 2015-07-27 at 20 18 05](https://cloud.githubusercontent.com/assets/340766/8962922/c32cc5f8-35f3-11e5-9bdc-c229892ecef8.png)

Haven't mentioned it before because I know you are not very fond of drastic changes, I hope it's useful at least as a reference.

Flaburgan commented:
Btw I toke a nice picture of a dandelion to use it as a background. I will put it under CC-BY ;)

svbergerem commented:
@pablocubico I'd like to have the header on all pages, I don't like those icons (that's why I removed them) and we will have to think about some good content for user and podmin oriented pages. 

And don't get me wrong. I like the things you changed. The points I listed are exactly those things that you didn't change and which I already dislike about the current landing page.

> I hope it's useful [...] as a reference.

It definitely is. The big picture at the top looks good and I am curious what the dandelion picture looks like that @Flaburgan took. Like the title of this PR already says: this is work in progress. I don't like our current landing page so I did some quick changes and pushed them so we can discuss how the new landing page should look like.

I think before we go into any details how the design should look like we should first define what kind of content we would like to put on those pages.

I like the idea of having two different pages: one for the podmin that we show right after the installation and a user oriented one. Maybe we add a new page for the podmin with a new url and just redirect to that page when there is just one user on the pod, no admin, or some other rule we will have to define. Then the podmin could still look up stuff even if the pod is already running for some time.

On the podmin page I think at least the following information would be useful:
* configure your pod
* create an account
* make yourself an admin
* updating (link to the wiki page)
* getting help (wiki: faq for pod maintainers, irc)
* contribute (issues, code)

Flaburgan commented:
Another way to check which page to display could also be to check the url: localhost or 127.0.0.1 -> podmin page (or even "developer" page).

pablocubico commented:
@svbergerem Totally agree with you, content is always first and it should come before design. You design for the content, I'm glad to see discussions about content on those pages.

As for my changes, my goal is to always reduce coding efforts to a minimum, so I preserved as much as I could, to also make use of the current language schemes. And of course it has my style, which is always personal.

Having a podmin page and a separate user page out of the box will be very useful.

Flaburgan commented:
![img_1059](https://cloud.githubusercontent.com/assets/930064/8965210/55e35cbe-362a-11e5-8e80-5c2509adf768.JPG)
![img_1069](https://cloud.githubusercontent.com/assets/930064/8965211/55e59c90-362a-11e5-899b-30ba8f2d876c.JPG)
![img_1070](https://cloud.githubusercontent.com/assets/930064/8965213/55eaff00-362a-11e5-9122-b2988390167d.JPG)
![img_1073](https://cloud.githubusercontent.com/assets/930064/8965214/55f141bc-362a-11e5-8b98-d63f73e7492f.JPG)
![img_1079](https://cloud.githubusercontent.com/assets/930064/8965311/facb2270-362a-11e5-9cf0-390ae82f9abb.JPG)
![img_1080](https://cloud.githubusercontent.com/assets/930064/8965312/fad020f4-362a-11e5-8ee0-2f9f9b7cbafc.JPG)
![img_1083](https://cloud.githubusercontent.com/assets/930064/8965313/fad1cf6c-362a-11e5-8bef-d54e3a3b2f73.JPG)
![img_1085](https://cloud.githubusercontent.com/assets/930064/8965314/fad49260-362a-11e5-98fe-d59e05ea28ea.JPG)

Here are some of the pictures. I toke them for https://github.com/diaspora/diaspora/pull/5389 that's why I placed the dandelion on the left. Those are raw files under CC-BY, I guess a Gimp master can improve the images. Resize / adapt as you want!

pablocubico commented:
@Flaburgan awesome pics!

svbergerem commented:
Here are some new screenshots:

### podmin page
![podmin page](https://cloud.githubusercontent.com/assets/3798614/8970041/9085c5e2-3647-11e5-8adb-9ac2002f3e43.png)
* small image below the header (CC-BY)
* information on the most important topics (unfinished, will be translatable)

### user page
![user page](https://cloud.githubusercontent.com/assets/3798614/8970272/0ced9500-3649-11e5-99e6-df9c9d8dee50.png)
* bigger image below the header
* „Welcome to %{podname}“
* some content (just basic information, currently missing, will be translatable)
* sign up / sign in buttons

jaywink commented:
I don't think maintaining a podmin page justifies the work required. Please, lets drop it and concentrate on the users? All podmins will go through the wiki or some other installation information page.

We can add podmin tips etc to the admin view instead of maintaining a separate landing page.

Would be awesome to get this default landing page renewed, totally like the one posted above by @svbergerem !

svbergerem commented:
@jaywink Like @Flaburgan wrote:

> Some pods are installed without following the install guide (with apt-get, with bitnami etc).

We are trying to make the installation as easy as possible and in the future there might be even more podmins who didn't follow the installation guides in our wiki. I think besides the UX we should also try to improve the PX. (podmin experience)

Try to imagine you are a podmin who just finished the installation. You hope everything went well and you would like to test your pod. You open the website in your browser and the first thing you see is a nice page which says

> Welcome, friend. You're about to change the Internet. Let's get you set up, shall we?

and below there are some guides for your first steps as a podmin.

jaywink commented:
So how do you determine when to show the user page and when to show the podmin page?

I just feel it's very little benefit for the extra code - 99% of podmins will hopefully know we have a wiki - or maybe if not we've failed.. :) But won't oppose if others feel the podmin landing page still should stay.

diaspora-code-review commented on app/views/home/podmin.haml:
```diff
@@ -0,0 +1,61 @@
+          .entypo-key
+          = t(".make_yourself_an_admin")
+        != t(".make_yourself_an_admin_info",
+             wiki: link_to("diaspora* wiki", "https://wiki.diasporafoundation.org/FAQ_for_pod_maintainers#What_are_roles_and_how_do_I_use_them.3F_.2F_Make_yourself_an_admin"),
```
Line is too long. [175/120]

jaywink commented:
> So how do you determine when to show the user page and when to show the podmin page?

Still wondering about this? ;)

kas70 commented:
https://github.com/diaspora/diaspora/pull/6268#issuecomment-126102251
this looks absolutely awesome. i'd like to see this in [0.6.0.0](https://github.com/diaspora/diaspora/milestones/0.6.0.0). can we add this to this milestone?

svbergerem commented:
The main problem with this PR is that I don't have any good ideas for the user page. I already posted a screenshots above but there is still no content.

Flaburgan commented:
>there is still no content

This is podmin job, to welcome his users and present his pod as he wants, isn't it?

So maybe we should only use the podmin welcome page and indicate to the podmin how to modify it to welcome his users.

jaywink commented:
Still wondering about this question which I've asked twice with no comments ;)

> So how do you determine when to show the user page and when to show the podmin page?

svbergerem commented:
@jaywink Let me [quote myself](https://github.com/diaspora/diaspora/pull/6268#issuecomment-126015137):

> I like the idea of having two different pages: one for the podmin that we show right after the installation and a user oriented one. Maybe we add a new page for the podmin with a new url and just redirect to that page when there is just one user on the pod, no admin, or some other rule we will have to define. Then the podmin could still look up stuff even if the pod is already running for some time.

So I'd say we check if there are at least 2 users and at least one of those has the admin role. In this case we would show the user page. Else we redirect to the podmin page.

jaywink commented:
Follow up question - why? A suddenly disappearing podmin info page sounds like a misfeature than a feature to me.

What about having the landing page always be the user info and have an addition link to url `/podmin` or something to show the podmin info page *when the podmin is logged in*.

The "single user info page" rule to me is confusing. What if when I set up a new public pod and I want to get 1000 users on it. Before anyone actually does, the podmin page will be shown? That just doesn't sound logical.

svbergerem commented:
@jaywink 
> Follow up question - why? A suddenly disappearing podmin info page sounds like a misfeature than a feature to me.

We can inform the podmin on this page that there will be another landing page once there is another user on their pod and they marked themselves as an admin. The page also won't “disappear”. If the podmin reloads the page they will still see the same page. They just won't be redirected there once a second user signed up.

> What about having the landing page always be the user info and have an addition link to url /podmin or something to show the podmin info page when the podmin is logged in.

We already have the /admin page. The purpose of the podmin _landing_ page is a different one: it should inform the podmin about the next steps they might want to take right after setting up the pod. 

> The "single user info page" rule to me is confusing. What if when I set up a new public pod and I want to get 1000 users on it. Before anyone actually does, the podmin page will be shown? That just doesn't sound logical.

If you set up a pod with 1000 users you might want to have a dedicated podmin/team account which will be added to the admin role. And in most cases you will also have a private account on your pod. As soon as you created those accounts every new user will see the user landing page.

If you don't want to have a private account on your pod then this PR still improves the situation. Currently we **always** show the podmin landing page because there is none for users.

svbergerem commented:
@Flaburgan 
> This is podmin job, to welcome his users and present his pod as he wants, isn't it?

Yeah, because we all know that most podmins change the landing page of their pod, right. ;-) I'd just like to have some okayish default landing page for users. Podmins can still use their own but the current default one is really bad.

svbergerem commented:
I added some code to redirect users to the podmin page in the cases I described above. I also added some “content” to the users page for testing purposes. Good ideas are still welcome!

![user page](https://cloud.githubusercontent.com/assets/3798614/12975015/cc2292e0-d0b8-11e5-878d-903882c6f549.png)

AugierLe42e commented:
I would add some parallax on the image. This kind of details always look nice. ;)

Commit:
Redesign default.haml and add a podmin welcome page

svbergerem commented:
### default.haml
![default haml](https://cloud.githubusercontent.com/assets/3798614/16180665/4f4a6e5c-368c-11e6-9e8b-15ebd1685a27.png)

### podmin.haml
![podmin haml](https://cloud.githubusercontent.com/assets/3798614/16180666/6020794c-368c-11e6-99cd-935866e80cdd.png)

denschub commented:
Pretty. I like it. Merged.

MightyCreak commented:
I'm always a bit confused with the "Sign up" / "Sign in" thing, English is not my first language and I honestly don't know which one is for creating a new account and which one is for logging in.

Personally I find "Log in" / "Create account" far more easier while being as relevant.

jhass commented:
We will perform browser language detection and use the terms in your native langugae.

MightyCreak commented:
Sure, but don't you think that even for a English native the "up" and "in" thing is a bit confusing?

I'm just saying it because I've read a UX article about that (in English) saying that it might be better for the users to have distinct words for each action. It's easier for the brain to choose quickly from one to another.

What I'm concerned for is that the same philosophy goes on with the translation, which means trying to find words that are graphically close but that mean different thing.

svbergerem commented:
@goobertron As a native speaker: what do you think about "Log in"/"Create account" or "Sign in"/"Create account" instead of our current "Sign in"/"Sign up"?

goobertron commented:
> As a native speaker: what do you think about "Log in"/"Create account" or "Sign in"/"Create account" instead of our current "Sign in"/"Sign up"?

I'm happy with any of the three. Sign in/Sign up is clear to me. If language detection will be performed even on the splash page when not signed out and no cookies stored, the second-language issue should no longer be a problem. However, I'm happy for it to be changed if this will help some people.

Personally I prefer 'sign in' to 'log in' as it sounds a bit less like a tech-admin term and more 'real-life' to me. So if you want to make the two terms more different, I would vote for "Sign in"/"Create account".

Thanks for asking me!

svbergerem commented:
Thank you @MightyCreak and @goobertron! #6881 uses “Create account” instead of “Sign up”.

MightyCreak commented:
Thank you for your understanding ;)

goobertron commented:
I had a thought last night about a possible alternative. 'Create account' seems quite functional and formal. As a lot of Diaspora's UX for new people tries to be less formal and more welcoming than many sites, how about 'Join us' instead of 'Create account'? Possibly even 'Join us!' with an exclamation mark to make it seem even more encouraging. So, 'Sign in / Join us!'. Or it could be 'Join diaspora*', although that could be confused with a certain pod... or 'Join the community', although this is quite long. I'm just making some suggestions in case you like one of them.
